.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,27 @@
+name: Run Benchmarks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Run Go Benchmark Action
+        uses: patrickhuie19/benchmark-action@b46d0ce6c3b4bd423380f7d3c6abed389e38ea4a
+        with:
+          benchmarks-pr: 'BenchmarkKeystore_Sign'  # Use benchmarks specified in PR, or default benchmarks
+          benchmarks-merge: 'BenchmarkKeystore_Sign' # Default list of benchmarks to run on merges
```

Comment:
Can we default to all benchmarks to avoid a maintenance burden?

Reply:
If someone writes a benchmark will they be comfortable with the performance data being stored and reporting on all subsequent commits that include that benchmark? 

It's not a one way door, the backing action can be updated whenever we like.